app/geo/gui/src/ts/twxplore/gui/geo/components/filterSliders/FilterComponent.tsx

```diff
@@ -0,0 +1,243 @@
+import * as React from "react";
+import {makeStyles, Theme, createStyles} from "@material-ui/core/styles";
+import Typography from "@material-ui/core/Typography";
+import Slider from "@material-ui/core/Slider";
+import {connect, useSelector, useDispatch} from "react-redux";
+import {MapState} from "../../states/map/MapState";
+import {RootState} from "../../states/root/RootState";
+import {setFilter} from "kepler.gl/actions";
+import {getFeatureAttributeByName} from "../../attributeStrategies/getFeatureAttributeByName";
+import {MapFeatureTypeState} from "../../states/map/MapFeatureTypeState";
+import {allFiltersSet} from "../../actions/map/AllFiltersSetAction";
+import {FeatureType} from "../../api/graphqlGlobalTypes";
+import {MapFeatureAttributeState} from "../../states/map/MapFeatureAttributeState/MapFeatureAttributeState";
+import {TypeOfFeatureAttribute} from "../../states/map/TypeOfFeatureAttribute";
+import {MapNumericFeatureAttributeState} from "../../states/map/MapFeatureAttributeState/MapNumericFeatureAttributeState";
+import {MapStringFeatureAttributeState} from "../../states/map/MapFeatureAttributeState/MapStringFeatureAttributeState";
+import {FormControl, TextField} from "@material-ui/core";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: 300,
+    },
+    margin: {
+      height: theme.spacing(3),
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120,
+      maxWidth: 300,
+    },
+    chips: {
+      display: "flex",
+      flexWrap: "wrap",
+    },
+    chip: {
+      margin: 2,
+    },
+    noLabel: {
+      marginTop: theme.spacing(3),
+    },
+  })
+);
+
+function valuetext(value: number) {
+  return `${value}`;
+}
+
+//var filterIndexOfAttribute: number | null; //necessary to pass this into onChangeCommitted since it
+const FilterSlidersImpl: React.FunctionComponent<{featureType: string}> = ({
+  featureType,
+}) => {
+  const classes = useStyles();
+  const state: MapState = useSelector(
+    (rootState: RootState) => rootState.app.map
+  );
+
+  const setInitialFilter = (
+    filterIndexOfAttribute: number,
+    attributeName: string,
+    stateOfAttribute: MapFeatureAttributeState
+  ) => {
+    /*dispatch(
+      setFilter(
+        filterIndexOfAttribute,
+        "domain",
+        getFeatureAttributeByName(attributeName).fieldType
+      )
+    );*/
+    dispatch(setFilter(filterIndexOfAttribute, "name", attributeName));
+    dispatch(
+      setFilter(
+        filterIndexOfAttribute,
+        "fieldType",
+        getFeatureAttributeByName(attributeName).fieldType
+      )
+    );
+    dispatch(
+      setFilter(
+        filterIndexOfAttribute,
+        "type",
+        getFeatureAttributeByName(attributeName).filterType
+      )
+    );
+    switch (getFeatureAttributeByName(attributeName).typeOf) {
+      case TypeOfFeatureAttribute.NUMBER: {
```

Comment:
I see how you're using this TypeOfFeatureAttribute, sortof.

This is really where the strategy pattern is supposed to shine. Right now you're using FeatureAttribute as a set of flags indicating how to switch on certain behavior. But the strategy pattern is really about polymorphic delegation. Can you turn this switch into a delegation to the strategy? i.e.,

const featureAttribute = getFeatureAttributeByName(attributeName);
featureAttribute.getInitialFilterActions(...)

You could have two levels of strategy singletons, like:

featureAttribute.filterType.getInitialFilterActions(...).

in order to reuse code.